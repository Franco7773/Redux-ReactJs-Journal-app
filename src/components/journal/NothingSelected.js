import React from 'react'

export const NothingSelected = () => {
  return (
    <div className="noting__main-content">
      <p>
        Select something
        <br />
        pr create an entry
      </p>

      <i className="far fa-star fa-4x mt_5"></i>
    </div>
  );
}
