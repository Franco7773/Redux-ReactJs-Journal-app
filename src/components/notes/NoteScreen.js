import React, { useEffect, useRef } from 'react'
import { NotesAppBar } from './NotesAppBar'
import { useSelector, useDispatch } from 'react-redux';
import { useForm } from '../../hooks/useForm';
import { activeNote, startDeleting } from '../../actions/notes';

export const NoteScreen = () => {

  const dispatch = useDispatch();
  const { active: note } = useSelector( state => state.notes );
  const [{ id, body, title, date, url }, handleInputChange, reset ] = useForm( note );
  const activeId = useRef( note.id );

  useEffect(() => {

    if (note.id !== activeId.current) {
      reset( note );
      activeId.current = note.id
    }
  }, [ reset, note ]);

  useEffect(() => {
  
    dispatch( activeNote( id, { id, title, body, date, url }));
  
  }, [ dispatch, id, title, body, date, url ]);
  
  const handleDelete = () => {
    dispatch( startDeleting( id ));
  };

  return (
    <div className="notes__main-content">
      <NotesAppBar />

      <div className="notes__content">
        <input value={ title } onChange={ handleInputChange } name="title" type="text" className="notes__content--title-input" autoComplete="off" placeholder="Some awesome title"></input>

        <textarea value={ body } onChange={ handleInputChange } name="body" className="notes__content--textarea" placeholder="What happened today?"></textarea>

        { url &&
          <div className="notes__content--image">
            <img alt="imagen" src={ url } />
          </div>
        }
      </div>

      <button onClick={ handleDelete } className="btn btn--danger">Delete</button>
      
    </div>
  );
}
