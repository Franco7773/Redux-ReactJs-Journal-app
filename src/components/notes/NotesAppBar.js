import React from 'react'
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {

  const dispatch = useDispatch();
  const { active } = useSelector( state => state.notes );

  const handleSave = () => {
    dispatch( startSaveNote( active ));
  };

  const handlePictureClick = () => {
    // dispatch( startSaveNote( active ));
    document.querySelector('#fileSelector').click();
  };

  const handleFileChange = ( e ) => {
    e.preventDefault();

    const file = e.target.files[0];

    if (file) {
      dispatch( startUploading( file ));
    }
  };
  
  return (
    <div className="notes__appbar">
      <span>27 de Agosto 2020</span>

      <input id="fileSelector" onChange={ handleFileChange } type="file" name="file" style={{ display: 'none' }}/>

      <div>
        <button onClick={ handlePictureClick } className="btn">Picture</button>
        <button onClick={ handleSave } className="btn">Save</button>
      </div>
    </div>
  );
}
