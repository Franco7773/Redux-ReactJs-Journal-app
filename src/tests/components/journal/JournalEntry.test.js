import React from 'react';
import { Provider } from 'react-redux';

import '@testing-library/jest-dom'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { mount } from 'enzyme';

import { JournalEntry } from '../../../components/journal/JournalEntry';
import { activeNote } from '../../../actions/notes';


jest.mock('../../../actions/notes', () => ({
  activeNote: jest.fn()
}));

const middlewares = [thunk];

const mockStore = configureStore( middlewares );

const initState = { };

const store = mockStore( initState );
store.dispatch = jest.fn();

const note = {
  id: 11,
  date: 0,
  title: 'Hello',
  body: 'Galaxy',
  url: 'https://google.com/photo.jpg'
};

const wrapper = mount(
  <Provider store={ store }>
    <JournalEntry {...note } />
  </Provider>
);

describe('Pruebas en <JournalEntry />', () => {
  
  test('Debe de mostrarse correctamente', () => {
    
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de activar la nota', () => {
    
    wrapper.find('.journal__entry').prop('onClick')();

    expect( store.dispatch ).toHaveBeenCalled(
      activeNote( note.id, {...note })
    );
  });
});
