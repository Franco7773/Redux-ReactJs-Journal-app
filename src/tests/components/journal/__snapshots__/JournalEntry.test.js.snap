// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <JournalEntry /> Debe de mostrarse correctamente 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry--picture"
    style={
      Object {
        "backgroundImage": "url(https://google.com/photo.jpg)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry--body"
  >
    <p
      className="journal__entry--body-tittle"
    >
      Hello
    </p>
    <p
      className="journal__entry--body-content"
    >
      Galaxy
    </p>
  </div>
  <div
    className="journal__entry--date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
